
Every kernel release there are a few changes to headers
made. Some code gets shifted around between headers or
new headers are defined. This patch deals with such
cases.

--- a/drivers/net/wireless/b43/phy_common.h
+++ b/drivers/net/wireless/b43/phy_common.h
@@ -2,6 +2,9 @@
 #define LINUX_B43_PHY_COMMON_H_
 
 #include <linux/types.h>
+#if (LINUX_VERSION_CODE == KERNEL_VERSION(2,6,28))
+#include <asm/atomic.h>
+#endif
 
 struct b43_wldev;
 
--- a/drivers/net/wireless/libertas/assoc.c
+++ b/drivers/net/wireless/libertas/assoc.c
@@ -5,6 +5,9 @@
 #include <linux/ieee80211.h>
 #include <linux/if_arp.h>
 #include <net/lib80211.h>
+#if (LINUX_VERSION_CODE < KERNEL_VERSION(2,6,29))
+#include <asm/unaligned.h>
+#endif
 
 #include "assoc.h"
 #include "decl.h"
--- a/drivers/net/wireless/wl12xx/wl1251_main.c	2009-08-07 11:42:12.532495607 -0700
+++ b/drivers/net/wireless/wl12xx/wl1251_main.c	2009-08-07 12:08:31.167548696 -0700
@@ -26,6 +26,9 @@
 #include <linux/firmware.h>
 #include <linux/delay.h>
 #include <linux/irq.h>
+#if (LINUX_VERSION_CODE == KERNEL_VERSION(2,6,28))
+#include <linux/device.h>
+#endif
 #include <linux/crc32.h>
 #include <linux/etherdevice.h>
 #include <linux/vmalloc.h>
--- a/drivers/net/wireless/wl12xx/wl1251_spi.c	2009-08-07 12:24:58.907526225 -0700
+++ b/drivers/net/wireless/wl12xx/wl1251_spi.c	2009-08-07 12:24:59.199494403 -0700
@@ -24,6 +24,9 @@
 #include <linux/irq.h>
 #include <linux/module.h>
 #include <linux/crc7.h>
+#if (LINUX_VERSION_CODE == KERNEL_VERSION(2,6,28))
+#include <linux/device.h>
+#endif
 #include <linux/spi/spi.h>
 #include <linux/spi/wl12xx.h>
 
--- a/net/mac80211/key.c
+++ b/net/mac80211/key.c
@@ -21,6 +21,9 @@
 #include "aes_ccm.h"
 #include "aes_cmac.h"
 
+#if (LINUX_VERSION_CODE < KERNEL_VERSION(2,6,29))
+#include <asm/unaligned.h>
+#endif
 
 /**
  * DOC: Key handling basics
--- a/drivers/net/wireless/iwlwifi/iwl-core.h	2009-12-10 22:29:01.000000000 -0800
+++ b/drivers/net/wireless/iwlwifi/iwl-core.h	2009-12-10 22:47:19.000000000 -0800
@@ -63,7 +63,11 @@
 #ifndef __iwl_core_h__
 #define __iwl_core_h__
 
+#if (LINUX_VERSION_CODE >= KERNEL_VERSION(2,6,33))
 #include <generated/utsrelease.h>
+#else
+#include <linux/utsrelease.h>
+#endif
 
 /************************
  * forward declarations *
